pkg/api/mlflow/service/run/service.go

```diff
@@ -520,11 +521,10 @@ func (s Service) LogParam(
 
 	param := convertors.ConvertLogParamRequestToDBModel(run.ID, req)
 	if err := s.paramRepository.CreateBatch(ctx, 1, []models.Param{*param}); err != nil {
-		if _, ok := err.(repositories.ParamConflictError); ok {
+		if errors.As(err, &repositories.ParamConflictError{}) {
```

Comment:
should this not be `errors.Is` instead so we don't use the unwrapped error below and potentially lose parts of the error stack?

Reply:
I think we have to change from Error interface to error instance for `Is` to work.

Reply:
I'd suggest we keep the error interface and `As` usage so that each ParamConflictError can carry the particular message.

Reply:
I looked into this further, and we could make `Is` work by implementing `func (e ParamConflictError) Is(target error) bool`. However this does not seem necessary (at least at this stage), as indeed `As` does what we want here. I originally thought that `err` would get unwrapped and changed by the call to `As`, but this is not the case. It's the second parameter that gets assigned, but you don't use it so all is good!